enhancements/node-latency-profile/node-latency-profiles.md

```diff
@@ -0,0 +1,276 @@
+	// +optional
+	Status NodeLatencyProfileStatus `json:"status"`
+}
+```
+
+Please refer to the `Kubespray` [documentation](https://github.com/kubernetes-sigs/kubespray/blob/master/docs/kubernetes-reliability.md#recommendations-for-different-cases) for more details on different scenarios.
+```go
+type ProfileType int
+
+const (
+    // Fast Update and Fast Reaction.
+    // See - https://github.com/kubernetes-sigs/kubespray/blob/af04906b518efe8d6dcfc9e2ff57b17b49c20295/docs/kubernetes-reliability.md#fast-update-and-fast-reaction
+    FUFR ProfileType = "FUFR" //
+
+    // Medium Update and Average Reaction
+    // See - https://github.com/kubernetes-sigs/kubespray/blob/af04906b518efe8d6dcfc9e2ff57b17b49c20295/docs/kubernetes-reliability.md#medium-update-and-average-reaction
+    MUAR ProfileType = "MUAR"
+
+    // Low Update and Slow Reaction
+    // See - https://github.com/kubernetes-sigs/kubespray/blob/af04906b518efe8d6dcfc9e2ff57b17b49c20295/docs/kubernetes-reliability.md#low-update-and-slow-reaction
+    LUSR ProfileType = "LUSR"
+)
+```
+
+```go
+type NodeLatencyProfileSpec struct {
+  // NodeLatencyProfileType determins the type of NodeLatencyProfile to set on the cluster.
+  // Only follow values "FUFR", "MUAR" or "LUSR" are allowed.
+	NodeLatencyProfileType ProfileType `json:"profileType"`
+}
+```
+```go
+// NodeLatencyProfileStatus is used by MCO, KCMO and CKAO for synchronization
+type NodeLatencyProfileStatus struct {
+    // KubeletUpdateStatus indicates the status of the kubelet arguments update rollout
+    KubeletUpdateStatus KubeletArgsUpdateStatus `json:"kubeletArgsupdateStatus"`
+
+    // ControllerManagerUpdateStatus indicates the status of the controller manager arguments update rollout
+    ControllerManagerUpdateStatus ControllerManagerArgsUpdateStatus `json:"controllerManagerUpdateStatus"`
+
+    // KubeAPIServerUpdateStatus indicates the status of the Kube APIServer arguments update rollout
+    KubeAPIServerUpdateStatus KubeAPIServerArgsUpdateStatus `json:"kubeAPIServerUpdateStatus"`
+
+    // NodeLatencyProfileUpdateStatus indicates overall progress of NodeLatencyProfile rollout
+    NodeLatencyProfileUpdateStatus ProfileUpdateStatus `json:"status"`
+}
+```
+More on those individual status types below,
+
+```go
+
+// CurrentState denotes the current state of the rollout of the NodeLatencyProfile
+type CurrentState string
+
+const (
+    NotStartedState CurrentState = "NotStarted"
+    InProgressState CurrentState = "InProgress"
+    CompleteState   CurrentState = "Complete"
+    FailedState     CurrentState = "Failed"
+    RollBackState   CurrentState = "RollBacked"
+)
+
+type ProfileUpdateStatus struct {
+    State   CurrentState `json:"currentState"`
+    Message string       `json:"message,omitempty"`
+}
+
+type KubeletArgsUpdateStatus struct {
+    // KubeletConfigName is the name of the KubeletConfig used for updating the kubelet arguments
+    KubeletConfigName string `json:"KubeletConfigName"`
```

Comment:
what is this and why does it matter?

Reply:
It has been documented in section `Coordination between MCO, KCMO, CKAO` subsection `MCO`. In short, MCO needs to know which `KubeletConfig` to track (out of many pre-existing) used for updating Kubelet flags relevant to the given `NodeLatencyProfile`